### DISCUSSION TITLE
@collection.* API Filter doesnt work

### DISCUSSION THREAD

**Discussion by ValentinKolb** [pocketbase/pocketbase#4928] [Q&A]
I have a complex API filter that should prevent a user from creating an entry in an entry collection if he already has an entry at the same time for the same event.

My tables are:

```
events: {
   id
}

eventLists: {
   id
   allowOverlappingEntries: boolean
   event: events relation
}

eventListSlots {
  id
  eventList: eventLists relation
  startDate: Date
  endDate: Date
}

eventListSlotEntries {
  id
  eventListSlot: eventListSlots relation
  user: users relation
}
```

This is my query: 

```
@request.auth.id != "" && (
  eventListsSlot.eventList.allowOverlappingEntries = true
  ||
  (
    @collection.eventListSlotEntries.eventListsSlot.eventList.event.id ?=
     eventListsSlot.eventList.event.id
    &&
    @collection.eventListSlotEntries.user ?= @request.auth.id
    &&
    (
      @collection.eventListSlotEntries.eventListsSlot.startDate ?> 
        eventListsSlot.endDate
      ||
      @collection.eventListSlotEntries.eventListsSlot.endDate ?<
        eventListsSlot.startDate
    )
  )
)
```

The Problem is, that this Filter won't let the User his first entry record (error: `DrySubmit create rule failure: sql: no rows in result set`) but if the user already has an entry, the filter lets everything through, even overlapping entries.

What am i missing?

**Comment by ganigeorgiev** [comment]
The `@collection` field performs a `JOIN` and it is working correctly for me.

Unfortunately I don't know what is the desired behavior in your case as there are not enough details about your requirements to understand or attempt to reproduce it locally.

Note that currently the Create API rule executes **after** a dry submit of the record, aka. your entry will be already available in the database for the API rule, aka. you can try to exclude the created record (eg. `id ?!= @collection.eventListSlotEntries.id`) if that's the issue in your case.

**Reply by ganigeorgiev** [reply]
Also the dates `>`, `<` comparisons should be without the `?` prefix if you want to target all joined  `eventListSlotEntries.eventListsSlot` entries (_similar to an SQL query, rules acts as a "funnel" where the results are getting filtered with each following constraint_).

**Reply by ValentinKolb** [reply]
That you very much for your quick reply!
